test/opentelemetry-metrics/fixtures.test.js

```diff
@@ -112,13 +112,13 @@ async function checkEventsHaveTestMetrics(t, events, extraMetricNames = []) {
     // A histogram that we expect to have the APM agent default buckets.
     m = event.metricset.samples.test_histogram_defbuckets;
     t.equal(m.type, 'histogram', 'test_histogram_defbuckets.type');
-    t.equal(m.counts.length, 3, 'test_histogram_defbuckets.counts');
+    t.equal(m.counts.length, 2, 'test_histogram_defbuckets.counts');
```

Comment:
@trentm why these values have changed?

Reply:
The test fixture adds three data points:

```
const histo = meter.createHistogram('test_histogram_defbuckets');
// ...
  histo.record(2);
  histo.record(3);
  histo.record(4);
```

Per https://github.com/elastic/apm/blob/main/specs/agents/metrics-otel.md#histogram-aggregation our default histogram bucket boundaries are:

```
[0.00390625, 0.00552427, 0.0078125, 0.0110485, 0.015625, 0.0220971, 0.03125,
0.0441942, 0.0625, 0.0883883, 0.125, 0.176777, 0.25, 0.353553, 0.5, 0.707107, 1,
1.41421, 2, 2.82843, 4, 5.65685, 8, 11.3137, 16, 22.6274, 32, 45.2548, 64,
90.5097, 128, 181.019, 256, 362.039, 512, 724.077, 1024, 1448.15, 2048, 2896.31,
4096, 5792.62, 8192, 11585.2, 16384, 23170.5, 32768, 46341.0, 65536, 92681.9,
131072]
```

So, before https://github.com/open-telemetry/opentelemetry-js/pull/4935 when the upper boundary was *exclusive* those data points (2, 3, and 4) feel in three buckets:

- `2` in `(2, 2.82843)`
- `3` in `(2.82843, 4)`
- `4` in `(4, 5.65685)`

After that sdk-metrics change, when the upper bounder is now *inclusive* they fall into *two* buckets:

- `2` in `(1.41421, 2]`
- `3` in `(2.82843, 4]`
- `4` also in `(2.82843, 4]`

Reply:
Thanks for the thorough explanation :)